### DISCUSSION TITLE
How to use setPathParams?

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#5175] [Q&A] [answered]
Hello,

I have a router in a JS hook and would like to modify the :key on the fly.

```js
routerUse((next) => {
    return (c) => {
        const key = ctx.pathParam('key'); // <-- how to modify key?
        return next(c)
    }
})
```

May I have an example on how to use the `setPathParams`?

Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
Even if you change the path parameter what is the desired behavior - to change the matching route or to continue matching the current one and just to replace the path param value that it will use from the handler?

If you are trying to modify before the route match then it needs to be registered using `routerPre` (it is the equivalent of [`echo.Pre`](https://pkg.go.dev/github.com/labstack/echo/v5#Echo.Pre)).

At the moment I don't think it is possible to use `setPathParams` from the JS hooks because we don't have a constructor/method that will create [`echo.PathParam`](https://pkg.go.dev/github.com/labstack/echo/v5#PathParam`). A workaround for this could be to fetch the current params, modify the value and then call `setPathParams` with the modified value.

```js
routerPre((next) => {
    return (c) => {
        const params = c.pathParams()

        for (let param of params) {
            if (param.name == "example") {
                param.value = "newValue"
            }
        }

        c.setPathParams(params)

        return next(c)
    }
})
```

**Reply by AfdhalAtiffTan** [reply]
Thank you for the super fast reply Gani!

I will give it a go tomorrow.

The desired outcome is to remove any invalid path.

I am working on #5139 and I want to sanitise the backup name.
